R/plot_taxa.R

```diff
@@ -173,48 +262,188 @@ plot_taxa_accum_time <- function(observation, id, alpha = 1) {
 #' @export
 #' 
 #' @examples
-#' observation <- ants_L1[[1]]$tables$observation
-#' id <- names(ants_L1)
+#' \dontrun{
+#' # plot richness through time for ecocomDP formatted dataset by 
+#' # observation date
+#' plot_taxa_richness(
+#'   dataset = ants_L1)
 #' 
-#' plot_taxa_diversity(observation, id)
+#' # plot richness through time for ecocomDP formatted dataset by 
+#' # aggregating observations within a year
+#' plot_taxa_richness(
+#'   dataset = ants_L1,
+#'   time_window_size = "year")
 #' 
-plot_taxa_diversity <- function(observation, id, alpha = 1) {
+#' # plot richness through time for ecocomDP observation table
+#' plot_taxa_richness(
+#'   observation = ants_L1[[1]]$tables$observation)
+#' 
+#' # plot richness through time for flattened ecocomDP dataset 
+#' plot_taxa_richness(
+#'   flat_data = flatten_dataset(ants_L1))
+#' 
+#' 
+#' # Using Tidy syntax:
+#' # plot ecocomDP formatted dataset richness through time by 
+#' # observation date
+#' ants_L1 %>% plot_taxa_richness()
+#' 
+#' ants_L1 %>% plot_taxa_richness(time_window_size = "day")
+#' 
+#' # plot ecocomDP formatted dataset richness through time 
+#' # aggregating observations within a month
+#' ants_L1 %>% plot_taxa_richness(time_window_size = "month")
+#' 
+#' # plot ecocomDP formatted dataset richness through time 
+#' # aggregating observations within a year
+#' ants_L1 %>% plot_taxa_richness(time_window_size = "year")
+#' 
+#' # tidy syntax, filter data by date
+#' ants_L1 %>% 
+#'   flatten_dataset() %>% 
+#'   dplyr::filter(
+#'     lubridate::as_date(datetime) > "2007-01-01") %>%
+#'   plot_taxa_richness(
+#'     flat_data = .,
+#'     time_window_size = "year")
+#' }
+plot_taxa_richness <- function(
```

Comment:
These changes create a backwards incompatibility.

The semantics between the existing and proposed function names seem equivalent (or nearly so). Rather than changing an exported function name, consider updating the function metadata title with the proposed new function name. Would this work?

Reply:
sure, "richness" is more precise language than "diversity", but in this case, we could keep the old fxn name. On the upside, that would allow us to add other diversity metrics/methods to this fxn in the future.

Reply:
Let's go with keeping the old name, possibly modifying the title listed in the docs, and allowing us to add other diversity metrics methods to this function in the future.